### PR TITLE
Consolidating user session retrieval methods

### PR DESCRIPTION
**Pull request by alxdarksage**
Which ensures 410/412s occur together, because logic isn't different. All the methods now use the same code. Addresses BRIDGE-1550, BRIDGE-1411.

### CODE REVIEW

File: app/org/sagebionetworks/bridge/play/controllers/SubpopulationController.java [hunk 1]
```diff
@@ -69,10 +63,8 @@ public Result getSubpopulation(String guid) {
         return okResult(subpop);
     }
     public Result deleteSubpopulation(String guid, String physicalDeleteString) {
-        UserSession session = getAuthenticatedSession();
-        if (!session.isInRole(DELETE_ROLES)) {
-            throw new UnauthorizedException();
-        }
+        UserSession session = getAuthenticatedSession(ADMIN, DEVELOPER);
```
Comment (zaclinsage) [inline app/org/sagebionetworks/bridge/play/controllers/SubpopulationController.java:66]: will this just throws an exception if user does not have admin or dev permission?

File: test/org/sagebionetworks/bridge/play/controllers/SurveyControllerTest.java [hunk 2]
```diff
@@ -132,11 +144,18 @@ private void setUserSession(String studyIdentifier) {
...
+        
+        Map<SubpopulationGuid,ConsentStatus> consentStatuses = Maps.newHashMap();
+        consentStatuses.put(guid, status);
+        
         session = new UserSession(participant);
         session.setStudyIdentifier(new StudyIdentifierImpl(studyIdentifier));
-        doReturn(session).when(controller).getAuthenticatedSession();
-        doReturn(session).when(controller).getAuthenticatedAndConsentedSession();
-        doReturn(session).when(controller).getAuthenticatedSession(any(Roles.class));
+        session.setConsentStatuses(consentStatuses);
```
Comment (zaclinsage) [inline test/org/sagebionetworks/bridge/play/controllers/SurveyControllerTest.java]: why we need to set consent statuses up while we didn't use it in previous test codes?

File: app/org/sagebionetworks/bridge/play/controllers/SubpopulationController.java (hunk 1 above)
Comment (alxdarksage) [reply app/org/sagebionetworks/bridge/play/controllers/SubpopulationController.java:66]: Yes

**Review by zaclinsage — APPROVED** [review]
looks good to me

**Pull request merged**